volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+
+        # Last displayed must be between -1 and max
+        if self.HistoryBufferCount < 1 or self.HistoryBufferCount > max_buffers:
+            return False
+
+        if not self.get_title() and not self.get_original_title():
+            return False
+
+        return True
+
+    def get_screens(self):
+        """Generator for screens in the console.
+
+        A console can have multiple screen buffers at a time,
+        but only the current/active one is displayed.
+
+        Multiple screens are tracked using the singly-linked
+        list _SCREEN_INFORMATION.Next.
+
+        See CreateConsoleScreenBuffer
+        """
+        screens = [self.CurrentScreenBuffer]
+
+        if self.ScreenBuffer not in screens:
+            screens.append(self.ScreenBuffer)
+
+        seen = set()
+
+        for screen in screens:
+            cur = screen
+            while cur and cur.vol.offset != 0 and cur.vol.offset not in seen:
+                cur.TextBufferInfo.BufferRows.Rows.count = (
+                    cur.TextBufferInfo.BufferCapacity
+                )
+                yield cur
+                seen.add(cur.vol.offset)
+                cur = cur.Next
+
+    def get_histories(self):
+        for cmd_hist in self.HistoryList.dereference().to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_COMMAND_HISTORY",
+            "ListEntry",
+        ):
+            yield cmd_hist
+
+    def get_exe_aliases(self):
+        exe_alias_list = self.ExeAliasList
+        # Windows 10 22000 and Server 20348 made this a Pointer
+        if isinstance(exe_alias_list, objects.Pointer):
+            exe_alias_list = exe_alias_list.dereference()
+        for exe_alias_list_item in exe_alias_list.to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_EXE_ALIAS_LIST",
+            "ListEntry",
+        ):
+            yield exe_alias_list_item
+
+    def get_processes(self):
+        for proc in self.ConsoleProcessList.dereference().to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_CONSOLE_PROCESS_LIST",
+            "ListEntry",
+        ):
+            yield proc
+
+    def get_title(self):
+        try:
+            return self.Title.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+        except:
+            return ""
+
+    def get_original_title(self):
+        try:
+            return self.OriginalTitle.dereference().cast(
+                "string", encoding="utf-16", errors="replace", max_length=512
+            )
+        except:
+            return ""
+
+
+class COMMAND(objects.StructType):
+    """A Command Structure"""
+
+    def is_valid(self):
+        if (
+            self.Length < 1
+            or self.Allocated < 1
+            or self.Length > 1024
+            or self.Allocated > 1024
+        ):
+            return False
+
+        return True
+
+    def get_command(self):
```

Comment:
This feels like the same method as `ALIAS.get_source()`?  Is there any way of making sure the two stay in sync?

Reply:
combined this in one place